core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -506,6 +518,12 @@ public static <E extends Entry<Key,Value>> TabletMetadata convertRow(Iterator<E>
     if (buildKeyValueMap) {
       te.keyValues = kvBuilder.build();
     }
+    te.filesSupplier = Suppliers.memoize(() -> {
+      ImmutableMap.Builder<StoredTabletFile,DataFileValue> fb = ImmutableMap.builder();
+      te.rawFiles.forEach(pair -> fb.put(new StoredTabletFile(pair.getFirst()),
+          new DataFileValue((pair.getSecond()))));
+      return fb.build();
```

Comment:
The Guava immutable types are nice APIs to work with, but for simple immutable collections that are constructed in one small part of the code, it's also pretty easy to use something like `.stream()...collect(Collectors.toUnmodifiableMap())` to avoid more dependence on Guava entirely. This might be easier if using a multimap for the source of the stream.

Here's some other ideas: https://www.baeldung.com/java-stream-immutable-collection

I just think using `.forEach(map.put)` is probably the less readable than `stream.collect(mapcollector)`. It's direct, and simple enough, as a programmer to understand.

Reply:
For this case the rest of the code is using immutable map builders and those make sense in the other parts of the code.  For this code it could use stream, but personally I would rather keep it consistent with the code in the immediate vicinity.

Reply:
For what it's worth, I think the other code could change, too. I seem to remember there was some resistance to getting rid of `ImmutableX.builder()` code when I last did a pass through the code to replace our use of immutable Guava types with builtin immutables. The builders are nice. Even if we keep them, I think they could be expressed as a custom stream Collector. But that change would be out of scope of this PR.

One other consideration we have to be careful about is checking for null values `.contains(null)` fails on built-in Java immutables, like that returned from `List.of()`, `Set.of()`, etc., instead of just always returning false, which would make more sense. I don't think Guava has the same issue. And, I know `Collections.unmodifiableX()` does not suffer from the same problem.

Reply:
> I seem to remember there was some resistance to getting rid of ImmutableX.builder()

If the rest of the code could be changed to use something built in with java and have the same or better performance I personally would not have  a problem with that.  That is out of scope for this PR as you said though.